std/assembly/string.ts

```diff
@@ -206,21 +206,44 @@ export class String {
     return this.indexOf(searchString, position) != -1;
   }
 
-  indexOf(searchString: String, position: i32 = 0): i32 {
+  indexOf(searchString: String, fromIndex: i32 = 0): i32 {
     assert(this !== null);
     if (searchString === null) searchString = changetype<String>("null");
-    var pos: isize = position;
+    if (!searchString.length) return 0;
```

Comment:
Could this instead use `searchLen` from below to avoid the additional load? Like moving the definition of `searchLen` a bit up, when `searchString` isn't null?

Reply:
Done!
Also make `changetype("null")` as global constant

Reply:
I believe that top-level constants are always compiled currently because the initializer might have side effects and thus become (possibly unused) static memory segments no matter whether used or not. A string literal, on the other hand, is only added when the compiler encounters it (multiple occasions of the same string literal are also deduplicated by the compiler).

Reply:
Hmm, so better leave as is? Also I thinking about add helper function like `clamp` for make code little bit clearer because I see a lot of min(max(a,b),c)

Reply:
I believe the ECMAScript spec uses that `min/max` terminology explicitly at some places, though not in `Array#indexOf` as it seems. From the WAT it appears that using min/max directly has a bit better untouched code generation, but if you think that clamp helps readability significantly, I am also not opposed to doing this. Wdyt?

Reply:
I can revert to min/max if this really matter, but I think performance should be the same. By the way it seems compiler can't inline generic functions without manual type assign i.e `clamp<isize>(...)` inlined but `clamp(...)` doesn't for some reason.